src/compiler/moduleSpecifiers.ts

```diff
@@ -456,7 +458,7 @@ function getLocalModuleSpecifier(moduleFileName: string, info: Info, compilerOpt
     const allowedEndings = getAllowedEndingsInPrefererredOrder(importMode);
     const relativePath = rootDirs && tryGetModuleNameFromRootDirs(rootDirs, moduleFileName, sourceDirectory, getCanonicalFileName, allowedEndings, compilerOptions) ||
         processEnding(ensurePathIsNonModuleName(getRelativePathFromDirectory(sourceDirectory, moduleFileName, getCanonicalFileName)), allowedEndings, compilerOptions);
-    if (!baseUrl && !paths || relativePreference === RelativePreference.Relative) {
+    if (!baseUrl && !paths && !getResolvePackageJsonImports(compilerOptions) || relativePreference === RelativePreference.Relative) {

```

Comment:
```suggestion
    if (!baseUrl && !paths && relativePreference === RelativePreference.Relative || !getResolvePackageJsonImports(compilerOptions)) {
```

Reply:
This suggestion would break auto-import for `paths` and `baseUrl` if you're not using a module resolution mode that allows `package.json#imports` resolution?

Reply:
aren't they logically equal? `... && a || b` `... && b || a`